views/wsprov_fetch_linked_objects.aql

```diff
@@ -4,12 +4,21 @@
 // Args:
 //   ws_ids - array of private workspace ids the user has access to
 //   obj_keys - wsprov_object key to find links for
+//   is_private - if present, limit to objects in workspaces that a user has access to
+//   is_public - if present, limit to objects in public workspaces
+//   *** if both is_private and is_public are true this will be treated as an OR ***
+//   types - simplified workspace types to return (e.g. Genome, Assembly)
 //   link_limit - number of link results (10-20 recommended)
 
-for o in wsprov_object
-  filter o._key in @obj_keys
-  for link in 1..10 any o wsprov_links options {uniqueVertices: 'global', bfs: true}
-    filter link.is_public || link.workspace_id IN @ws_ids
-    limit @link_limit
-    collect parent_id = o._id into links = link
-    return { links, parent_id }
+FOR o IN wsprov_object
+  FILTER o._key in @obj_keys
+  FOR v IN 1..10 ANY o wsprov_links
+    OPTIONS {uniqueVertices: 'global', bfs: true}
+    LET simple_type = SPLIT(SPLIT(v.ws_type, '-', 1)[0], '.')[1]
```

Comment:
Not saying we need it, but we can register a stored function for this: https://docs.arangodb.com/3.4/AQL/Extending/

Reply:
Interesting. seems like there are some tradeoffs with that approach WRT indexing

Reply:
I don't think indexing would be used with an expression like that, unless we make a separate, indexed "simple_type" column. But I find that arango queries are still pretty fast without using indexes